lib/setup/setupInstall.js

```diff
@@ -1116,13 +1115,21 @@ function Install(options) {
      */
     this.enumerateAdapters = tools.poorMansAsync(function* (knownObjIDs, adapter) {
         let resultCode = 0;
+
+        const adapterRegex1 = new RegExp(`^system\\.adapter\\.${adapter}$`);
+        const adapterRegex2 = new RegExp(`^system\\.adapter\\.${adapter}\\.`);
+
         try {
-            const doc = yield getObjectViewAsync('system', 'adapter', {startkey: 'system.adapter.' + adapter, endkey: 'system.adapter.' + adapter + '\u9999'});

```

Comment:
I'm not sure here... Is there any case where multiple objects with type `adapter` can even exist? Otherwise we can greatly simplify this method.

Reply:
You are right. And the view expression is anyway wrong. It will return e.g. the admin and admin2. So we must simplify it.

Reply:
I've changed the method, please take another look.

Reply:
As I see it, "should not" :-) But exactly @GermanBluefox needs to say